rust/lance-core/src/utils/tracing.rs

```diff
@@ -75,4 +75,11 @@ pub const IO_TYPE_OPEN_MEM_WAL: &str = "open_mem_wal_index";
 pub const IO_TYPE_LOAD_VECTOR_PART: &str = "load_vector_part";
 pub const IO_TYPE_LOAD_SCALAR_PART: &str = "load_scalar_part";
 pub const TRACE_EXECUTION: &str = "lance::execution";
+pub const TRACE_DATASET_EVENTS: &str = "lance::dataset_events";
```

Comment:
I think all these new contents need to be after `EXECUTION_PLAN_RUN`, so that it is after everything about `TRACE_EXECUTION`